### PR TITLE
Add pages directories

### LINKED ISSUE #5735: Proposal to identify "page-level" components
We do have some components that are "page-level" components because we do not instantiate them ourselves, we pass them to React Router, which does the instantiation and so on.

Those components should be identified as "page-level" components. Those components get props from the React Router layer (without a HOC) and we should make it clear that they are a bit special.

I see two options:

1. we introduce a new directory `pages` either in `src/<project>/` or `src/<project>/components/`
2. we suffix these components with `Page`, but I don't really like the idea

I'd go for a new directory, probably: `src/<project>/components/pages/`.

What do you think?

### PR DESCRIPTION
**Pull request by willdurand**
Fix #5735 

---

### CODE REVIEW

File: tests/unit/amo/components/pages/TestCollectionList.js [hunk 1]
```diff
@@ -1,10 +1,10 @@
-import { oneLineTrim } from 'common-tags';
+import { oneLine } from 'common-tags';
```
Comment (rebmullin) [inline tests/unit/amo/components/pages/TestCollectionList.js]: did you mean to update this here?

**Comment by willdurand** [comment]
> I think I'm personally more of a fan of `src/<project>/pages/` - just from a “it’s easier to spot” point of view. I feel like it could a little lost with all the components in each project. I guess as long as it’s separated for the reasons you mentioned in the issue, then this will work 👍

the thing is those "things" are still components.

**Comment by willdurand** [comment]
> Quick question - should App stay outside?

I think so yeah, it has no "route".

**Comment by rebmullin** [comment]
>  think so yeah, it has no "route".

I know ..it just thought since it was a wrapper I wasn't sure if it should look "special" too : )

File: src/amo/components/AddonsCard/index.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ import type {
   DeleteAddonNoteFunc,
   RemoveCollectionAddonFunc,
   SaveAddonNoteFunc,
-} from 'amo/components/Collection';
+} from 'amo/components/pages/Collection';
```
Comment (kumar303) [inline src/amo/components/AddonsCard/index.js]: I'd prefer `amo/pages/Collection`. I get that they are technically components but I just don't want to go digging for them -- there are enough component sub-directories as it is. I'd rather see it moved up a level.

File: src/amo/components/App/index.js [hunk 3]
```diff
@@ -31,7 +31,7 @@ import {
   VIEW_CONTEXT_HOME,
   maximumSetTimeoutDelay,
 } from 'core/constants';
-import DefaultErrorPage from 'core/components/ErrorPage';
+import DefaultErrorPage from 'core/components/pages/ErrorPage';
```
Comment (kumar303) [inline src/amo/components/App/index.js]: I don't think `ErrorPage` makes sense in `pages`. I know that there are routes for them but they are just debugging routes used for development. I see `NotAuthorized`, `NotFound`, and so on as low-level components that routed pages will import and render.

File: src/disco/components/Routes/index.js [hunk 4]
```diff
@@ -3,12 +3,12 @@ import config from 'config';
 import * as React from 'react';
 import { Route, Switch } from 'react-router-dom';
 
-import GenericError from 'core/components/ErrorPage/GenericError';
-import NotFound from 'core/components/ErrorPage/NotFound';
-import DiscoPane from 'disco/components/DiscoPane';
-import SimulateAsyncError from 'core/components/error-simulation/SimulateAsyncError';
-import SimulateClientError from 'core/components/error-simulation/SimulateClientError';
-import SimulateSyncError from 'core/components/error-simulation/SimulateSyncError';
+import GenericError from 'core/components/pages/ErrorPage/GenericError';
```
Comment (kumar303) [inline src/disco/components/Routes/index.js]: Again, I don't think these components make sense in `pages`.

File: src/amo/components/AddonsCard/index.js (hunk 2 above)
Comment (willdurand) [reply src/amo/components/AddonsCard/index.js]: Let's do this!

**Pull request merged**